test_matrix.py

```diff
@@ -0,0 +1,57 @@
+import unittest
+from Matrix import Matrix
+
+
+class Test_matrix(unittest.TestCase):
```

Comment:
When there is a lot of logic, it might be better to split to several testCases in a way that each case handle one type of actions